scripts/qa_crowbarsetup.sh

```diff
@@ -1956,23 +1956,60 @@ function proposal_set_value
     proposal_modify_value "$1" "$2" "$3" "$4" "="
 }
 
+# get the currently set value in the proposal
+function proposal_get_value
+{
+    local proposal="$1"
+    local proposaltype="$2"
+    local variable="$3"
+
+    local pfile=`get_proposal_filename "${proposal}" "${proposaltype}"`
+
+    safely rubyjsonparse "puts j${variable}" < $pfile
+}
+
 # wrapper for proposal_modify_value
 function proposal_increment_int
 {
     proposal_modify_value "$1" "$2" "$3" "$4" "+="
 }
 
+function setup_trusted_cert
+{
+    local certfile=$1
+    local keyfile=$2
+
+    if [[ $hacloud = 1 ]] ; then
+        cluster_node_assignment
+        local clusternodes_var=$(echo clusternodes${clusternameservices})
+        local nodes=${!clusternodes_var}
+        local cn=$(get_cluster_vip_hostname $clusternameservices)
+    else
+        local nodes=($(get_all_discovered_nodes | head -n 1))
+        local cn=$nodes
+    fi
+
+    openssl req -x509 -newkey rsa:1024 -keyout $cn.key -out $cn.pem \
+        -days 365 -nodes \
+        -subj "/C=US/ST=Denial/L=Springfield/O=Dis/CN=${cn}"
+
+    local node
+    for node in $nodes; do
+        ssh $node mkdir -p $(dirname $certfile)
+        ssh $node mkdir -p $(dirname $keyfile)
+        scp $cn.pem $node:$certfile
+        scp $cn.key $node:$keyfile
+        ssh $node cp $certfile /etc/pki/trust/anchors/${cn}.pem
```

Comment:
my secret plan of world domination would have been that the provisioner actually sets that up for us automatically . I have a very unfinished patch for that

Reply:
I don't know what your patch looks like but I think it would be hard to implement for production. You'd have to automate around submitting a CSR to some external CA. Copying the cert directly into the trust bundle is an easy hack for CI.

Reply:
it doesn't have to be external, the CA just has to be imported as trusted (same like you do it here right now)